src/deepEqual.js

```diff
@@ -20,7 +20,24 @@
  * deepEqual({test: {abc: 5}}, {test: {abc: 5, def: 4}}) === false
  */
 function deepEqual(a, b) {
-  // write code here
+  let paramType = typeof a && typeof b;
+  if (a === b) {
+    return true;
+  } else if (a && b && paramType === 'object') {
+    if (Object.keys(a).length === Object.keys(b).length) {
+      for (let key in a) {
+        if (key in b) {
+          if (!deepEqual(a[key], b[key])) {
+            return false;
+          }
+        } else {
+          return false;
+        }
+      }
+      return true;
+    }
```

Comment:
figure out how to reduce level of nesting in this function. currently it's 4.
You can do 3 easily here. maybe even 2.